customcommands/api.go

```diff
@@ -0,0 +1,129 @@
+package customcommands
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/botlabs-gg/yagpdb/v2/common/config"
+	"github.com/botlabs-gg/yagpdb/v2/customcommands/models"
+	"github.com/botlabs-gg/yagpdb/v2/premium"
+	"github.com/botlabs-gg/yagpdb/v2/web"
+	"github.com/volatiletech/sqlboiler/queries/qm"
+	"goji.io"
+
+	"goji.io/pat"
+)
+
+var (
+	confCCAPIEnabled        = config.RegisterOption("yagpdb.custom_command_api.enabled", "Enable custom command management API", false)
+	confCCAPIRequirePremium = config.RegisterOption("yagpdb.custom_command_api.require_premium", "Require guild to be premium for API access", true)
+)
+
+func (p *Plugin) initAPI() {
+	if !confCCAPIEnabled.GetBool() {
+		logger.Warn("Custom commands API disabled, not starting it")
+		return
+	}
+
+	logger.Info("Starting custom commands API")
+
+	subMux := goji.SubMux()
+	subMux.Use(canUseAPIMW)
+
+	// TODO: This needs better middleware, during development we don't care right now
+	subMux.Use(web.RequireActiveServer)
+
+	web.ServerPublicAPIMux.Handle(pat.New("/customcommands"), subMux)
+	web.ServerPublicAPIMux.Handle(pat.New("/customcommands/*"), subMux)
+
+	subMux.Handle(pat.Get(""), web.APIHandler(handleGetAllCustomCommands))
+	subMux.Handle(pat.Get("/"), web.APIHandler(handleGetAllCustomCommands))
+
+	subMux.Handle(pat.Get("/:id"), web.APIHandler(handleGetCustomCommand))
+	subMux.Handle(pat.Get("/:id/"), web.APIHandler(handleGetCustomCommand))
+}
+
+// TODO: maybe just don't care and send the full object
+// **OR**
+// TODO: figure out how to marshal(?) the model.CustomCommand into this.
+
+// APICustomCommand is a representation of a custom command for use with the public API.
+// It is essentially a trimmed-down version of the customcommands.CustomCommand struct,
+// serving as a wrapper around the same.
+type APICustomCommand struct {
+	ID            int64              `json:"id"`
+	Name          string             `json:"name" valid:",0,100"`
+	IsEnabled     bool               `json:"is_enabled"`
+	TriggerType   CommandTriggerType `json:"trigger_type"`
+	Trigger       string             `json:"trigger" valid:",0,1000"`
+	CaseSensitive bool               `json:"case_sensitive"`
+	Responses     []string           `json:"responses" valid:"template,10000"`
+}
```

Comment:
`valid` tags aren't necessary here.

Reply:
Honestly, I'm not even sure whether this whole struct is needed. There's a few questions I have to solve, the main one is how to serialise the `models.CustomCommand` struct into the above.

I'll remove the tags in the coming commit(s) and see how to proceed from there.